### PR TITLE
Add deployment connection, airflow-variable, and pool commands

### PR DESCRIPTION
**Pull request by sunkickr**
## Description

Add deployment connection, airflow-variable, and pool commands to manage a Deployment's airflow objects

adds a list, create, update, and copy command for each one. copy can copy airflow objects from deployment to another

## 🎟 Issue(s)

- https://github.com/astronomer/astro-cli/issues/1157

## 🧪 Functional Testing

manually tested

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cmd/cloud/deployment_objects.go [hunk 1]
```diff
@@ -0,0 +1,645 @@
...
+
+	return cmd
+}
+
+func newDeploymentAirflowVariableRootCmd(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "airflow-variable",
+		Aliases: []string{"var"},
+		Short:   "Manage deployment's airflow variables",
+		Long:    "Manage airflow variables stored in an Astro Deployment's metadata database.",
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:141]:
```suggestion
		Long:    "Manage Airflow variables stored in an Astro Deployment's metadata database.",
```

File: cmd/cloud/deployment_objects.go [hunk 2]
```diff
@@ -0,0 +1,645 @@
...
+	)
+	return cmd
+}
+
+func newDeploymentAirflowVariableListCmd(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"li"},
+		Short:   "list a Deployment's airflow variables",
+		Long:    "list airflow variables stored in an Astro Deployment's Airflow metadata database",
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:157]:
```suggestion
		Short:   "list a Deployment's Airflow variables",
		Long:    "list Airflow variables stored in an Astro Deployment's metadata database",
```

File: cmd/cloud/deployment_objects.go [hunk 3]
```diff
@@ -0,0 +1,645 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deploymentAirflowVariableList(cmd, out)
+		},
+	}
+	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "The deployment id to list airflow variables from.")
+	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "Name of the deployment to list airflow variables from")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:163]:
```suggestion
	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "The ID of the Deployment.")
	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "The name of the Deployment.")
```

File: cmd/cloud/deployment_objects.go [hunk 4]
```diff
@@ -0,0 +1,645 @@
...
+		Long:    "Create airflow variables for an Astro Deployment",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deploymentAirflowVariableCreate(cmd, out)
+		},
+	}
+	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "The deployment id to create airflow variables for")
+	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "Name of the deployment to create airflow variables for")
+	cmd.Flags().StringVarP(&varValue, "value", "v", "", "The airflow variables's value. Required to create an airflow variable")
+	cmd.Flags().StringVarP(&key, "key", "k", "", "The airflow variables's key. Required to create an airflow variable")
+	cmd.Flags().StringVarP(&description, "description", "", "", "The airflow variable's description")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:183]:
```suggestion
	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "The ID of the Deployment.")
	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "The name of the Deployment.")
	cmd.Flags().StringVarP(&varValue, "value", "v", "", "The Airflow variable value. Required.")
	cmd.Flags().StringVarP(&key, "key", "k", "", "The Airflow variable key. Required.")
	cmd.Flags().StringVarP(&description, "description", "", "", "The Airflow variable description.")
```

File: cmd/cloud/deployment_objects.go [hunk 5]
```diff
@@ -0,0 +1,645 @@
...
+	return cmd
+}
+
+//nolint:dupl
+func newDeploymentAirflowVariableUpdateCmd(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "update",
+		Aliases: []string{"up"},
+		Short:   "Update airflow variables for a Deployment",
+		Long:    "Update airflow variables for an Astro Deployment",
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:194]:
```suggestion
		Short:   "Update Airflow variables for a Deployment",
		Long:    "Update Airflow variables for an Astro Deployment",
```

File: cmd/cloud/deployment_objects.go [hunk 6]
```diff
@@ -0,0 +1,645 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deploymentAirflowVariableUpdate(cmd, out)
+		},
+	}
+	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "The deployment id to update airflow variables for")
+	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "Name of the deployment to update airflow variables for")
+	cmd.Flags().StringVarP(&varValue, "value", "v", "", "The airflow variables's value")
+	cmd.Flags().StringVarP(&key, "key", "k", "", "The airflow variables's key. Required to update an airflow variable")
+	cmd.Flags().StringVarP(&description, "description", "", "", "The airflow variable's description")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:203]:
```suggestion
	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "The ID of the Deployment.")
	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "The name of the Deployment.")
	cmd.Flags().StringVarP(&varValue, "value", "v", "", "The Airflow variable value. Required.")
	cmd.Flags().StringVarP(&key, "key", "k", "", "The Airflow variable key. Required.")
	cmd.Flags().StringVarP(&description, "description", "", "", "The Airflow variable description.")
```

File: cmd/cloud/deployment_objects.go [hunk 7]
```diff
@@ -0,0 +1,645 @@
...
+	cmd := &cobra.Command{
+		Use:     "copy",
+		Aliases: []string{"cp"},
+		Short:   "Copy the airflow variables from one Deployment to another",
+		Long:    "Copy airflow variables from one Astro Deployment to another Astro Deployment. If a variable already exits with same Key it will be updated",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deploymentAirflowVariableCopy(cmd, out)
+		},
+	}
+	cmd.Flags().StringVarP(&fromDeploymentID, "source-id", "s", "", "The deployment id to copy airflow variables from.")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:219]:
```suggestion
	cmd.Flags().StringVarP(&fromDeploymentID, "source-id", "s", "", "The ID of the Deployment to copy Airflow variables from.")
```

File: cmd/cloud/deployment_objects.go [hunk 8]
```diff
@@ -0,0 +1,645 @@
...
+	cmd.Flags().StringVarP(&fromDeploymentName, "source-name", "n", "", "Name of the deployment to copy airflow variables from")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:220]:
```suggestion
	cmd.Flags().StringVarP(&fromDeploymentName, "source-name", "n", "", "Name of the deployment to copy airflow variables from")
```
```suggestion
	cmd.Flags().StringVarP(&fromDeploymentName, "source-name", "n", "", "The of the Deployment to copy Airflow variables from")
```

File: cmd/cloud/deployment_objects.go [hunk 9]
```diff
@@ -0,0 +1,645 @@
...
+	cmd.Flags().StringVarP(&toDeploymentID, "target-id", "t", "", "The id of the deployment where copied airflow variables are created or updated")
+	cmd.Flags().StringVarP(&toDeploymentName, "target-name", "", "", "Name of the deployment where copied airflow variables are created or updated")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:222]:
```suggestion
	cmd.Flags().StringVarP(&toDeploymentID, "target-id", "t", "", "The ID of the Deployment to receive the copied Airflow variables")
	cmd.Flags().StringVarP(&toDeploymentName, "target-name", "", "", "The name of the Deployment to receive the copied Airflow variables")
```

File: cmd/cloud/deployment_objects.go [hunk 10]
```diff
@@ -0,0 +1,645 @@
...
+		Long:    "Update pools for an Astro Deployment",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deploymentPoolUpdate(cmd, out)
+		},
+	}
+	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "The deployment id to update pools for")
+	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "Name of the deployment to update pools for")
+	cmd.Flags().StringVarP(&name, "name", "", "", "The pools's value")
+	cmd.Flags().IntVarP(&slots, "slots", "s", 0, "The pools's key. Required to update an pool")
+	cmd.Flags().StringVarP(&description, "description", "", "", "The pool's description")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:294]:
```suggestion
	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "The ID of the Deployment,")
	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "The name of the Deployment.")
	cmd.Flags().StringVarP(&name, "name", "", "", "The pool value")
	cmd.Flags().IntVarP(&slots, "slots", "s", 0, "The pool key. Required.")
	cmd.Flags().StringVarP(&description, "description", "", "", "The pool description.")
```

File: cmd/cloud/deployment_objects.go [hunk 11]
```diff
@@ -0,0 +1,645 @@
...
+	cmd.Flags().StringVarP(&toDeploymentName, "target-name", "", "", "Name of the deployment where copied airflow variables are created or updated")
+
+	return cmd
+}
+
+func newDeploymentPoolRootCmd(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "pool",
+		Aliases: []string{"pl"},
+		Short:   "Manage deployment's pools",
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:231]: Please use the same conventions here as I suggested for previous commands

File: cmd/cloud/deployment_objects.go [hunk 12]
```diff
@@ -0,0 +1,645 @@
...
+
+	airlfowURL := fmt.Sprintf("%v", value)
+
+	return deployment.ConnectionList(airlfowURL, airflowAPIClient, out)
+}
+
+func deploymentConnectionCreate(cmd *cobra.Command, out io.Writer) error {
+	ws, err := coalesceWorkspace()
+	if err != nil {
+		return errors.Wrap(err, "failed to find a valid workspace")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:342]:
```suggestion
		return errors.Wrap(err, "failed to find a valid Workspace")
```

File: cmd/cloud/deployment_objects.go [hunk 13]
```diff
@@ -0,0 +1,645 @@
...
+	}
+
+	// Silence Usage as we have now validated command input
+	cmd.SilenceUsage = true
+
+	requestedField = requestString
+	// get Airflow URl
+	value, err := inspect.ReturnSpecifiedValue(ws, deploymentName, deploymentID, astroClient, requestedField)
+	if err != nil {
+		return errors.Wrap(err, "failed to find a deployments airflow webserver URL")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:331]:
```suggestion
		return errors.Wrap(err, "failed to find the Deployment Airflow webserver URL")
```

File: cmd/cloud/deployment_objects.go [hunk 14]
```diff
@@ -0,0 +1,645 @@
...
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:352]:
```suggestion
		return errors.Wrap(err, "failed to find the Deployment Airflow webserver URL")
```

File: cmd/cloud/deployment_objects.go [hunk 15]
```diff
@@ -0,0 +1,645 @@
...
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:382]:
```suggestion
		return errors.Wrap(err, "failed to find the Deployment Airflow webserver URL")
```

File: cmd/cloud/deployment_objects.go [hunk 16]
```diff
@@ -0,0 +1,645 @@
...
+		return errors.Wrap(err, "failed to find a valid workspace")
+	}
+
+	// Silence Usage as we have now validated command input
+	cmd.SilenceUsage = true
+
+	requestedField = requestString
+	// get To Airflow URls
+	if fromDeploymentName == "" && fromDeploymentID == "" {
+		fmt.Println("Which deployment should connections be copied from?")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:408]:
```suggestion
		fmt.Println("Which Deployment should connections be copied from?")
```

File: cmd/cloud/deployment_objects.go [hunk 17]
```diff
@@ -0,0 +1,645 @@
...
+	}
+
+	return deployment.ConnectionUpdate(airlfowURL, connID, connType, description, host, login, password, schema, extra, port, airflowAPIClient, out)
+}
+
+//nolint:dupl
+func deploymentConnectionCopy(cmd *cobra.Command, out io.Writer) error {
+	ws, err := coalesceWorkspace()
+	if err != nil {
+		return errors.Wrap(err, "failed to find a valid workspace")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:399]:
```suggestion
		return errors.Wrap(err, "failed to find a valid Workspace")
```

File: cmd/cloud/deployment_objects.go [hunk 18]
```diff
@@ -0,0 +1,645 @@
...
+	cmd.SilenceUsage = true
+
+	requestedField = requestString
+	// get To Airflow URls
+	if fromDeploymentName == "" && fromDeploymentID == "" {
+		fmt.Println("Which deployment should connections be copied from?")
+	}
+	fromValue, err := inspect.ReturnSpecifiedValue(ws, fromDeploymentName, fromDeploymentID, astroClient, requestedField)
+	if err != nil {
+		return errors.Wrap(err, "failed to find the source deployments airflow webserver URL")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:412]:
```suggestion
		return errors.Wrap(err, "failed to find the source Deployment Airflow webserver URL")
```

File: cmd/cloud/deployment_objects.go [hunk 19]
```diff
@@ -0,0 +1,645 @@
...
+	}
+
+	fromAirlfowURL := fmt.Sprintf("%v", fromValue)
+	if toDeploymentName == "" && toDeploymentID == "" {
+		fmt.Println("Which deployment should connections be pasted to?")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:417]:
```suggestion
		fmt.Println("Which Deployment should receive the connections?")
```

File: cmd/cloud/deployment_objects.go [hunk 20]
```diff
@@ -0,0 +1,645 @@
...
+	}
+	value, err := inspect.ReturnSpecifiedValue(ws, toDeploymentName, toDeploymentID, astroClient, requestedField)
+	if err != nil {
+		return errors.Wrap(err, "failed to find a deployments airflow webserver URL")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:421]:
```suggestion
		return errors.Wrap(err, "failed to find the Deployment Airflow webserver URL")
```

File: cmd/cloud/deployment_objects.go [hunk 21]
```diff
@@ -0,0 +1,645 @@
...
+
+	toAirlfowURL := fmt.Sprintf("%v", value)
+
+	return deployment.CopyConnection(fromAirlfowURL, toAirlfowURL, airflowAPIClient, out)
+}
+
+func deploymentAirflowVariableList(cmd *cobra.Command, out io.Writer) error {
+	ws, err := coalesceWorkspace()
+	if err != nil {
+		return errors.Wrap(err, "failed to find a valid workspace")
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:432]:
```suggestion
		return errors.Wrap(err, "failed to find a valid Workspace")
```

File: cmd/cloud/deployment_objects.go [hunk 22]
```diff
@@ -0,0 +1,645 @@
...
+	}
+
+	// Silence Usage as we have now validated command input
+	cmd.SilenceUsage = true
+
+	requestedField = requestString
+	// get Airflow URl
+	value, err := inspect.ReturnSpecifiedValue(ws, deploymentName, deploymentID, astroClient, requestedField)
```
Comment (jwitz) [inline cmd/cloud/deployment_objects.go:440]: Please update these details for the rest of the error messages:

- Capitalize Deployment, Workspace, and Airflow in all uses.
- Don't use possessive language for Deployments (ie. do not write "Deployment's)
- Which deployment should X be pasted to? => Which Deployment should receive X?

**Review by kushalmalani — APPROVED** [review]

File: cmd/cloud/deployment_objects.go [hunk 23]
```diff
@@ -0,0 +1,645 @@
...
+		return errors.Wrap(err, "failed to find a valid Workspace")
+	}
+
+	// Silence Usage as we have now validated command input
+	cmd.SilenceUsage = true
+
+	requestedField = requestString
+	// get To Airflow URls
+	if fromDeploymentName == "" && fromDeploymentID == "" {
+		fmt.Println("Which Deployment should connections be copied from?")
```
Comment (neel-astro) [inline cmd/cloud/deployment_objects.go:408]: what do we do in this case? should we return an error to specify either one and not both or prioritize one over the other

File: cmd/cloud/deployment_objects.go [hunk 24]
```diff
@@ -0,0 +1,645 @@
...
+	}
+	fromValue, err := inspect.ReturnSpecifiedValue(ws, fromDeploymentName, fromDeploymentID, astroClient, requestedField)
+	if err != nil {
+		return errors.Wrap(err, "failed to find the source Deployment Airflow webserver URL")
+	}
+
+	fromAirlfowURL := fmt.Sprintf("%v", fromValue)
+	if toDeploymentName == "" && toDeploymentID == "" {
+		fmt.Println("Which Deployment should receive the connections?")
```
Comment (neel-astro) [inline cmd/cloud/deployment_objects.go:417]: what do we do in this case? should we return an error to specify either one and not both or prioritize one over the other

File: cmd/cloud/deployment_objects.go [hunk 25]
```diff
@@ -0,0 +1,645 @@
...
+		return errors.Wrap(err, "failed to find a valid workspace")
+	}
+
+	// Silence Usage as we have now validated command input
+	cmd.SilenceUsage = true
+
+	requestedField = requestString
+	// get To Airflow URls
+	if fromDeploymentName == "" && fromDeploymentID == "" {
+		fmt.Println("Which deployment should airflow variables be copied from?")
```
Comment (neel-astro) [inline cmd/cloud/deployment_objects.go:519]: same here

File: cmd/cloud/deployment_objects.go [hunk 26]
```diff
@@ -0,0 +1,645 @@
...
+	}
+	fromValue, err := inspect.ReturnSpecifiedValue(ws, fromDeploymentName, fromDeploymentID, astroClient, requestedField)
+	if err != nil {
+		return errors.Wrap(err, "failed to find the source deployments airflow webserver URL")
+	}
+
+	fromAirlfowURL := fmt.Sprintf("%v", fromValue)
+	if toDeploymentName == "" && toDeploymentID == "" {
+		fmt.Println("Which deployment should airflow variables be pasted to?")
```
Comment (neel-astro) [inline cmd/cloud/deployment_objects.go:528]: same here

File: cmd/cloud/deployment_objects.go [hunk 27]
```diff
@@ -0,0 +1,645 @@
...
+		return errors.Wrap(err, "failed to find a valid workspace")
+	}
+
+	// Silence Usage as we have now validated command input
+	cmd.SilenceUsage = true
+
+	requestedField = requestString
+	// get To Airflow URls
+	if fromDeploymentName == "" && fromDeploymentID == "" {
+		fmt.Println("Which deployment should pools be copied from?")
```
Comment (neel-astro) [inline cmd/cloud/deployment_objects.go:626]: same here

File: cmd/cloud/deployment_objects.go [hunk 28]
```diff
@@ -0,0 +1,645 @@
...
+	}
+	fromValue, err := inspect.ReturnSpecifiedValue(ws, fromDeploymentName, fromDeploymentID, astroClient, requestedField)
+	if err != nil {
+		return errors.Wrap(err, "failed to find the source deployments airflow webserver URL")
+	}
+
+	fromAirlfowURL := fmt.Sprintf("%v", fromValue)
+	if toDeploymentName == "" && toDeploymentID == "" {
+		fmt.Println("Which deployment should pools be pasted to?")
```
Comment (neel-astro) [inline cmd/cloud/deployment_objects.go:635]: same here

File: cloud/deployment/inspect/inspect.go [hunk 29]
```diff
@@ -181,6 +181,36 @@ func getAdditional(sourceDeployment *astro.Deployment) map[string]interface{} {
...
+	// create a map for deployment.configuration
+	deploymentConfigMap = getDeploymentConfig(&requestedDeployment)
+	// create a map for deployment.alert_emails, deployment.worker_queues and deployment.astronomer_variables
+	additionalMap = getAdditional(&requestedDeployment)
+	// create a map for the entire deployment
+	printableDeployment = getPrintableDeployment(deploymentInfoMap, deploymentConfigMap, additionalMap)
+
+	value, err = getSpecificField(printableDeployment, requestedField)
+	if err != nil {
+		return err, nil
```
Comment (neel-astro) [inline cloud/deployment/inspect/inspect.go:209]:
```suggestion
		return nil, err
```
fixing typo

**Review by neel-astro — COMMENTED** [review]
Left few minor comments, looks ok otherwise

**Pull request merged**